src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1512,6 +1532,358 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
+
+        readonly_sas_token = cloud_storage_client.generate_shared_access_signature(
+            'b',
+            'sco',
+            'rl',
+            datetime.datetime.utcnow() + datetime.timedelta(days=1))
+
+        readonly_sas_token = readonly_sas_token.strip('?')
+    readonly_sas_token = readonly_sas_token.strip('?')
+
+    from knack.prompting import prompt_y_n
+
+    print()
+    print('This will deploy a daemon set to your cluster to collect logs and diagnostic information and '
+          f'save them to the storage account '
+          f'{colorama.Style.BRIGHT}{colorama.Fore.GREEN}{storage_account_name}{colorama.Style.RESET_ALL} as '
+          f'outlined in {format_hyperlink("http://aka.ms/AKSPeriscope")}.')
+    print()
+    print('If you share access to that storage account to Azure support, you consent to the terms outlined'
+          f' in {format_hyperlink("http://aka.ms/DiagConsent")}.')
+    print()
+    if not prompt_y_n('Do you confirm?', default="n"):
+        return
+    print()
+    print("Getting credentials for cluster %s " % name)
+    path=os.path.join(os.path.expanduser('~'), '.kube', 'config')
+    # Setting kubeconfig
+    # Send cloud information to telemetry
+    send_cloud_telemetry(cmd)
+    kube_config = set_kube_config(kube_config)
+    print()
+    print("Starts collecting diag info for cluster %s " % name)
+
+    sas_token = sas_token.strip('?')
+
+    deployment_yaml = urlopen(
+        "https://raw.githubusercontent.com/Azure/aks-periscope/latest/deployment/aks-periscope.yaml").read().decode()
+    deployment_yaml = deployment_yaml.replace("# <accountName, base64 encoded>",
+                                              (base64.b64encode(bytes(storage_account_name, 'ascii'))).decode('ascii'))
+    deployment_yaml = deployment_yaml.replace("# <saskey, base64 encoded>",
+                                              (base64.b64encode(bytes("?" + sas_token, 'ascii'))).decode('ascii'))
+    container_logs = "azure-arc"
+    kube_objects = "azure-arc/pod azure-arc/service azure-arc/deployment"
+    yaml_lines = deployment_yaml.splitlines()
+    for index, line in enumerate(yaml_lines):
+        if "DIAGNOSTIC_CONTAINERLOGS_LIST" in line and container_logs is not None:
+            yaml_lines[index] = line + ' ' + container_logs
+        if "DIAGNOSTIC_KUBEOBJECTS_LIST" in line and kube_objects is not None:
+            yaml_lines[index] = line + ' ' + kube_objects
+        if "CLUSTER_TYPE" in line:
+            yaml_lines[index] = '  CLUSTER_TYPE: connectedCluster'
+
+    deployment_yaml = '\n'.join(yaml_lines)
+
+    fd, temp_yaml_path = tempfile.mkstemp()
+    temp_yaml_file = os.fdopen(fd, 'w+t')
+    try:
+        temp_yaml_file.write(deployment_yaml)
+        temp_yaml_file.flush()
+        temp_yaml_file.close()
+        print(temp_yaml_file)
+        try:
+            kube_config = None
+            print("Cleaning up aks-periscope resources if existing")
```

Comment:
Maybe "Cleaning up diagnostic container resources from the k8s cluster if existing"